bpf/flows.c

```diff
@@ -89,13 +89,24 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
         aggregate_flow->dns_record.latency = pkt.dns_latency;
         aggregate_flow->dns_record.errno = dns_errno;
         long ret = bpf_map_update_elem(&aggregated_flows, &id, aggregate_flow, BPF_ANY);
-        if (trace_messages && ret != 0) {
+        if (ret != 0) {
+            u32 *error_counter_p = NULL;
+            u32 initval = 1, key = 0;
             // usually error -16 (-EBUSY) is printed here.
             // In this case, the flow is dropped, as submitting it to the ringbuffer would cause
             // a duplicated UNION of flows (two different flows with partial aggregation of the same packets),
             // which can't be deduplicated.
             // other possible values https://chromium.googlesource.com/chromiumos/docs/+/master/constants/errnos.md
-            bpf_printk("error updating flow %d\n", ret);
+            if (trace_messages) {
+                bpf_printk("error updating flow %d\n", ret);
+            }
+            // Update global counter for hashmap update errors
+            error_counter_p = bpf_map_lookup_elem(&global_counters, &key);
```

Comment:
to make sure if I understand that: you created `global_counters` has a map for generic purpose, ie. today it contains drop counters at key 0 but potentially later we may add more counters at different indexes, is this correct?
Perhaps if so you could define a constant DROP_COUNTER_KEY = 0 or something like that?

Reply:
yes and I am already thinking of using this to count filtered flows :) at different index as well as ovs dbg I will add const here and the golang side Thanks